devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/GradleUtils.java

```diff
@@ -58,4 +64,13 @@ public static List<Dependency> listProjectBoms(Project project) {
         return boms;
     }
 
+    @Nullable
+    public static JavaToolchainSpec getExecutableOverrideToolchainSpec(ObjectFactory objectFactory) {
+        String customExecutable = objectFactory.newInstance(DefaultJavaExecSpec.class).getExecutable();
```

Comment:
Just passing by:
FYI: JavaExec is doing that because for JavaExec you can set java as executable. For your tasks that doesn't make a lot of sense and `objectFactory.newInstance(DefaultJavaExecSpec.class).getExecutable();` will always return null. Also you are referencing internal type, it would be good great if you could avoid that :)

I assume you want to achieve logic like:
```
if (quarkusTask has set toolchain) {
   use that
} else {
  use toolchains from JavaExtension
} 
```

You can achieve that with something like:
```
Provider<JavaLauncher> javaExtensionToolchainLauncher = getToolchainService().launcherFor(getJavaPluginExtension().getToolchain());
task.getJavaLauncher().convention(javaExtensionToolchainLauncher);
```

See also https://docs.gradle.org/current/userguide/toolchains.html#sec:plugins_toolchains

I hope that helps! 👋